Revision session 1.1 and 1.2 (#47)
carpentries-incubator/r-geospatial-urban: episodes/02-data-structures.Rmd

You can create a vector with a `c()` function. 

You can also inspect vectors with `str()` function. In factor vectors, 
it shows the underlying values of each category. 
You can also see the structure in the environment tab of RStudio.

```{r vectors}

# vector of numbers - numeric data type.
numeric_vector <- c(2, 6, 3) 
numeric_vector
str(numeric_vector)

> # vector of words - or strings of characters- character data type. Note that we need to use '' to tell R that we are working with strings. If there is a ' mark in the string itselfs, such as s'Gravenhage, then we use ""

Reviewer: ```suggestion
# vector of words - or strings of characters- character data type. Note that we need to use quotation marks '' to tell R that we are working with strings. If there is a ' mark in the string itself, such as s'Gravenhage, then we use "".
```

Reviewer: It's a matter of convention I guess, but in DCHSS, we teach it's a good practice to consistently use double quotation marks, as single ones are more often found in strings.  Either way, might be good to move this information to callout?

Reviewer 2: Issue to be solved in https://github.com/carpentries-incubator/r-geospatial-urban/issues/70

> character_vector <- c('Amsterdam', "'s Gravenhage", 'Delft') 

Reviewer: well, again matter of convention, but personally, I would prefer to see all strings with double quotation marks

Reviewer 2: To be addressed as part of #70

> character_vector <- c('Amsterdam', "'s Gravenhage", 'Delft') 

Reviewer 2: Dropping this suggestions to keep the issue clean for #70. To be solved there.

```
combined_vector <- c(abcd_vector, numeric_vector)

> str(combined_vector)

Reviewer: ```suggestion
combined_vector
str(combined_vector)
```

Reviewer: For consistency with previous examples, I would print out vector before showing its structure

# stating if a statement 'This element of the vector is a missing value'
# is true or not

#in order to see how many values are missing in our with_na vector, we can use the sum function
sum(is.na(with_na))

> #if we want to identify the values that are not missing we write the following

Reviewer: ```suggestion
# to identify the values that are not missing we write the following
```

!is.na(with_na) # The ! operator means negation, i.e. not is.na(with_na)

> #and if we want to sum all the non-missing values we write

Reviewer: ```suggestion
#and to sum all the non-missing values we write
```

nordic_cat
```

> ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

Reviewer: in the present form, two callouts were merged to one.  I would probably keep them separate as one refers to reordering, other the vector values. 

If we want to keep 2 callouts, this line needs to be reincluded

Reviewer 2: Issue to be solved in https://github.com/carpentries-incubator/r-geospatial-urban/issues/71

You can also inspect vectors with `str()` function. In factor vectors, 
it shows the underlying values of each category. 
You can also see the structure in the environment tab of RStudio.

```{r factor-reorder}
str(nordic_cat) 
```

> :::::::::::::::::::::::::::::::::::::::::::::::::::: callout 

Reviewer: to keep two callouts, this line needs to be reincluded

Reviewer 2: @alwil see my suggestion below to split the callouts into two

Reviewer 2: To be solved as part of #71

nordic_cat
```

> ### Note of caution 

Reviewer 2: ```suggestion
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

:::::::::::::::::::::::::::::::::::::::::::::::::::: callout

### Note of caution 
```
